### ISSUE TITLE
Incorrect information on commiter

### ISSUE THREAD

**Issue by wingechr** [spapas/python-git-info#2]
I really like the tool, but for me, it shows the wrong name for the commiter. This is not a problem with the script, but rather the information stored in `.git/logs/refs/...`

This only seems to show the name of the account of the local checkout, and differs from the output of `$ git log` 

Any idea where that information is actually stored?

Cheers.

**Comment by spapas** [comment]
Hello @wingechr unfortunately i'm not sure I know what you mean :| Is there a repository somewhere (maybe on github?) I could clone and test this a bit myself ? 

Thanks !

**Comment by wingechr** [comment]
Thanks for the quick reply.

Here is a short example (I am using Debian):

```bash
$ git clone https://github.com/kubernetes-client/python.git kybernetes
$ cd kybernetes
$ git log -n1
```

This shows that the last commit `b519de15c27037ce71d432ec068ad10cc2e0a7a8` was made by 
`Author: Kubernetes Prow Robot <k8s-ci-robot@users.noreply.github.com>`

In the same directory i am using python3:

```bash
$ python3
```
```python
>>> from gitinfo import get_git_info
>>>  get_git_info()
{'commit_time': '2019-07-22 18:16:21', 'gitdir': '/home/c/kybernetes/.git', 'message': 'clone: from https://github.com/kubernetes-client/python.git', 'parent_commit': '0000
000000000000000000000000000000000000', 'commit': 'b519de15c27037ce71d432ec068ad10cc2e0a7a8', 'commiter': 'Christian Winger <c.winger@mydomain.com>'}
```
The commit hash is correct, but the commiter is my name/mail

This information comes from `.git/logs/refs/heads/master`

Is this the same for you?

Cheers

**Comment by spapas** [comment]
Hello @wingechr yes it is the same for me. It seems that when there's a clone or a fast-foward merge you won't get the same information as `git log`. I'll need to research that a bit to understand how `git log` actually works in order to extract that information. 

Unless you have a better idea of course :)

**Comment by wingechr** [comment]
Not really, sorry, but I have a suspicion that these information are stored in the `.idx` or `.pack` files, which are binary (at least `$ strace git log` shows that those are being read when using ` git log` )

**Comment by spapas** [comment]
Hello @wingechr I did some research and it seems that each commit message is *also* saved as an object in the `.git/objects` folder (for example the commit message/info for the commit `33bba5e893986797fd68c4515bfafd709c6f69e5` will be in the folder `/objects/33` in a file named `bba5e893986797fd68c4515bfafd709c6f69e5` compressed with zlib).

I've fixed the project (v 0.5) to parse this file to retrieve the information and it now seems to be working ok for cloned or merged projects. 

However this change opened a very large bucket of snakes: packed repositories. It seems that for large git projects git does not store each file individual but packages everything in a .pack file (and an accompanying .idx file). So I'll need to search inside these files to actually retrieve the commit message. I'm in the process of implementing it (see #3) but until I fix it it won't work properly with package repos :|

I'm closing this issue (since it should be fixed now at least for non-packages repos) in favor of #3.
